lib/ansible/modules/cloud/amazon/s3_bucket.py

```diff
@@ -259,7 +271,7 @@ def bucket_exists(s3_client, bucket_name):
 def create_bucket(s3_client, bucket_name, location):
     try:
         configuration = {}
-        if location not in ('us-east-1', None):
+        if location not in ('us-east-1', 'nyc3', 'ams3', 'sgp1', None):
```

Comment:
This is... not ideal.  I did this to get around the fact that DigitalOcean doesn't seem to like adding the `LocationContraint` configuration parameter to CreateBucket requests.  I'm open to other ways of handling this.  This line will have to be updated if and when DO adds new regions with Spaces support.

Reply:
probably doing a condition to verify if **module._name** is **digital_ocean_space** will be better otherwise, from your comment, if DO adds more regions this will fail.